Tests/SwiftParserTest/Assertions.swift

```diff
@@ -578,6 +611,76 @@ extension ParserTestCase {
+      fixItsApplications: FixItsApplication(applyFixIts: applyFixIts, expectedFixedSource: expectedFixedSource).map {
+        [$0]
+      } ?? [],
+      options: options,
+      swiftVersion: swiftVersion,
+      experimentalFeatures: experimentalFeatures,
+      file: file,
+      line: line
+    )
+  }
+
+  /// Verifies that parsing of `markedSource` produces expected results using a
+  /// combination of various testing techniques:
+  ///
+  /// 1. Asserts that parsing of `markedSource` round-trips, ie. that the printed
+  ///    parsed tree is the same as the input.
+  /// 2. Checks that parsing produces the expected list of diagnostics. If no
+  ///    diagnostics are passed, asserts that the input parses without any errors.
+  /// 3. Checks that applying fix-its of the source code results in the
+  ///    expected fixed source, effectively testing the Fix-Its.
+  /// 4. If a substructure is passed, asserts that the parsed tree contains a
+  ///    subtree of that structure.
+  /// 5. Mutates the test input by flipping each token's presence (ie. for every
+  ///    token, either remove it from the input if it is present in the parsed
+  ///    tree or synthesize it if it was missing) and verifies that this
+  ///    mutated input round-trips. This test is disabled if the
+  ///    `SKIP_LONG_TESTS` environment variable is set.
+  /// 6. If swift-syntax is compiled with the
+  ///    `SWIFTPARSER_ENABLE_ALTERNATE_TOKEN_INTROSPECTION` environment variable
+  ///    set, mutates the input based on tokens the parse is trying to parse.
+  ///    See the *Test Case Mutation* section in CONTRIBUTING.md.
+  ///
+  ///
+  /// - Parameters:
+  ///   - markedSource: source that can include markers of the form `1️⃣`,
+  ///     to be used as locations in the following parameters.
+  ///   - parse: The function with which the source code should be parsed.
+  ///     Defaults to parsing as a source file.
+  ///   - expectedSubstructure: Asserts the parsed syntax tree contains this structure.
+  ///   - substructureAfterMarker: Changes the position to start the structure
+  ///     assertion from, ie. allows matching a particular substructure rather
+  ///     than the whole source.
+  ///   - expectedDiagnostics: Asserts the given diagnostics were output, by default it
+  ///     asserts the parse was successful (ie. it has no diagnostics). Note
+  ///     that `DiagnosticsSpec` uses the location marked by `1️⃣` by default.
+  ///   - fixItsApplications: A list of `FixItsApplication` to check for whether applying certain fix-its to the source
+  ///     will generate a certain expected fixed source. An empty list means no fix-its are expected.
+  ///   - swiftVersion: The version of Swift using which the file should be parsed.
+  ///      Defaults to the latest version.
+  ///   - experimentalFeatures: A list of experimental features to enable, or
+  ///     `nil` to enable the default set of features provided by the test case.
```

Comment:
Instead of duplicating the comment (which will inevitably go out-of-sync). I would prefer to have one canonical version of the comment. The other `assertParse` function could then have a comment like

```swift
/// Shorthand version of ``assertParse-<someHash from docc>`` that allows checking a single configuration of applied Fix-Its.
```

Reply:
I couldn't get docc build for test targets, would there be any other way to get the hash?

Reply:
The only I know how to get them is to use code completion in Xcode.

Reply:
> ```swift
> /// Shorthand version of ``assertParse-<someHash from docc>`` that allows checking a single configuration of applied Fix-Its.
> ```

Could I just omit the the hash for now and go for the full name of `assertParse` instead?

Reply:
Ah, I forgot that `assertParse` is an internal function and not `public`, hence a docc link won’t work. 

So just use 
```
`assertParse`
```
with single backticks and make it a normal code font.